### PR TITLE
Add basic travis config and atomicfile.SaveFile support

### PR DESCRIPTION
**Pull request by stephen-soltesz**
This change adds a local package to support saving GCS Objects locally via `temp.SaveFile`.

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/gcs-downloader/1)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Travis configuration for gcs-downloader.
+
+language: go
+dist: xenial
+
+before_install:
+# Coverage tools
+- go get github.com/mattn/goveralls
+- go get github.com/wadey/gocovmerge
```
Comment (pboothe) [inline .travis.yml]: Don't need this

File: temp/file.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// Package temp defines file operations for safely writing files.
```
Comment (pboothe) [inline temp/file.go]: `temp` is a bad name. Name the package after what it does, not the technique it uses. `safewrite`, `atomicfile`, or something else are all better than `temp`.

**Review by pboothe — APPROVED** [review]
LGTM after travis cleanup and package rename.

**Review by stephen-soltesz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/gcs-downloader/1)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @pboothe)

___
*[.travis.yml, line 9 at r1](https://reviewable.io/reviews/m-lab/gcs-downloader/1#-LcIDKZdyupWF-TKArQu-r1-9:-LcIGD1m6ResbnJhhmfW:b-896fix) ([raw file](https://github.com/m-lab/gcs-downloader/blob/492de2e30e9cf18f03bea417ad9bbd2f525be28d/.travis.yml#L9)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Don't need this
</blockquote></details>

Done.
___
*[temp/file.go, line 1 at r1](https://reviewable.io/reviews/m-lab/gcs-downloader/1#-LcIDKZdyupWF-TKArQv-r1-1:-LcIGE7t5U2yOcgzqdcG:b12s2oq) ([raw file](https://github.com/m-lab/gcs-downloader/blob/492de2e30e9cf18f03bea417ad9bbd2f525be28d/temp/file.go#L1)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

`temp` is a bad name. Name the package after what it does, not the technique it uses. `safewrite`, `atomicfile`, or something else are all better than `temp`.
</blockquote></details>

Updated to `atomicfile`.

**Pull request merged**